docs/transaction.md

```diff
@@ -0,0 +1,261 @@
+# Life of a transaction
+
+This document describes how Dragonfly transactions provide atomicity and serializability for its multi-key and multi-command operations.
```

Comment:
DragonflyDB transaction algorithm provide linearizability and serializability

Reply:
what is atomicity of multi-key command, isnt it the same as Serializability?

Reply:
No, the atomicity of a multi-key command is called strict serializability. See at the bottom example of serializable but not linearizable schedule.